File: lista/Lista Dinâmica/Java/Lista Genérica/Listavel.java
public interface Listavel<T> {
	
	//anexa(insere ao final da lista) a lista o novo dado fornecido
	void anexar(T dado) throws OverflowException;
	
	//insere o novo dado fornecido na lista, numa posicao logica informada 
	void inserir(int posicao, T dado) throws OverflowException;
	
	//retorna o elemento que está numa posicao logica informada
	T selecionar(int posicao);
	
	//retorna o elemento que está numa posicao logica informada
	T[] selecionarTodos();

	//substitui o elemento de uma posicao logica informada,
	//pelo novo elemento fornecido
	void atualizar(int posicao, T novoDado);
	
	//remove o elemento de uma posicao logica informada
	T apagar(int posicao)throws UnderflowException;
	
	//métodos auxiliares
	boolean estaCheia();
	boolean estaVazia();
	String imprimir();
}

File: lista/dinamica/Kotlin/Lista Genérica/Listavel.kt
interface Listavel<T> {
    // Metodos principais
    fun anexar(dado: T?)                  //C
    fun inserir(posicao: Int, dado: T?)   //C
    fun selecionar(posicao: Int): T?      //R
    fun selecionarTodos(): Array<T?>      //R
    fun atualizar(posicao: Int, dado: T?) //U
    fun apagar(posicao: Int): T?          //D
    fun apagarTodos(): Array<T?>          //D
    fun limpar()                            //D
    
	//fun juntar(lista1: Listavel, lista2: Listavel): Listavel
	//fun subLista(inicio: Int, fim: Int): Listavel
    // Metodos auxiliares
    fun tamanho(): Int  
    fun estaCheia(): Boolean
    fun estaVazia(): Boolean
    fun imprimir(): String
}